app/controllers/locations_controller.rb

```diff
@@ -15,7 +15,149 @@ def index
+        render :name, status: :unprocessable_entity
+      end
+    end
+  end
+
+  def units
+    if params[:location].present?
+      @location.units = params[:location][:units]
+      if @location.valid?(:units)
+        @location.save!
+        if params[:referrer] == "check_answers"
+          redirect_to scheme_location_check_answers_path(@scheme, @location, route: params[:route])
+        else
+          redirect_to scheme_location_type_of_unit_path(@scheme, @location, route: params[:route])
+        end
+      else
+        render :units, status: :unprocessable_entity
+      end
+    end
+  end
+
+  def type_of_unit
+    if params[:location].present?
+      @location.type_of_unit = params[:location][:type_of_unit]
+      if @location.valid?(:type_of_unit)
+        @location.save!
+        if params[:referrer] == "check_answers"
+          redirect_to scheme_location_check_answers_path(@scheme, @location, route: params[:route])
+        else
+          redirect_to scheme_location_mobility_standards_path(@scheme, @location, route: params[:route])
+        end
+      else
+        render :type_of_unit, status: :unprocessable_entity
+      end
+    end
+  end
+
+  def mobility_standards
+    if params[:location].present?
+      @location.mobility_type = params[:location][:mobility_type]
+      if @location.valid?(:mobility_type)
+        @location.save!
+        if params[:referrer] == "check_answers"
+          redirect_to scheme_location_check_answers_path(@scheme, @location, route: params[:route])
+        else
+          redirect_to scheme_location_availability_path(@scheme, @location, route: params[:route])
+        end
+      else
+        render :mobility_standards, status: :unprocessable_entity
+      end
+    end
+  end
+
+  def availability
+    if params[:location].present?
+      day = params[:location]["startdate(3i)"]
+      month = params[:location]["startdate(2i)"]
+      year = params[:location]["startdate(1i)"]
+      if [day, month, year].none?(&:blank?)
+        if Date.valid_date?(year.to_i, month.to_i, day.to_i)
+          @location.startdate = Time.zone.local(year.to_i, month.to_i, day.to_i)
+          if @location.valid?(:startdate)
+            @location.save!
+            redirect_to scheme_location_check_answers_path(@scheme, @location, route: params[:route])
+          else
+            render :availability, status: :unprocessable_entity
+          end
+        else
+          error_message = I18n.t("validations.location.startdate_invalid")
+          @location.errors.add :startdate, error_message
+          render :availability, status: :unprocessable_entity
+        end
+      else
+        error_message = I18n.t("validations.location.startdate_blank")
+        @location.errors.add :startdate, error_message
+        render :availability, status: :unprocessable_entity
+      end
+    end
+  end
+
+  def check_answers
+    if params[:location].present?
+      @location.confirmed = true
+      @location.save!
```

Comment:
having discussed with the design team - I think this is the right place to mark the location as confirmed